### PR TITLE
Express some higher order functions in terms of other

### PR DESCRIPTION
**Pull request by mapogolions**

### CODE REVIEW

File: lib/array.js [hunk 1]
```diff
@@ -366,45 +366,72 @@ const series = (items, fn, done) => {
 //       accepted - <boolean>
 //   done - <Function>, on done
 //     err - <Error> | <null>
-//     result - <any>
+//     index - <int>
```
Comment (SemenchenkoVitaliy) [inline lib/array.js]:
```suggestion
//     index - <number>
```

Comment (mapogolions) [reply lib/array.js]: Thanks for feedback

File: test/array.findIndex.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+const metasync = require('..');
+const metatests = require('metatests');
+
+metatests.test('findIndex with error', test => {
```
Comment (DzyubSpirit) [inline test/array.findIndex.js]: Tests are good 👍 
Just nitpicking a grammar, not that important ;-)
```suggestion
metatests.test('findIndex with an error', test => {
```

File: test/array.findIndex.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    process.nextTick(() => callback(null, item % 3 === 0 && item % 5 === 0));
+
+  metasync.findIndex(data, predicate, (err, index) => {
+    test.error(err, 'must not return an error');
+    test.strictSame(index, expected, `result should be: ${expected}`);
+    test.end();
+  });
+});
+
+metatests.test('with empty array', test => {
```
Comment (DzyubSpirit) [inline test/array.findIndex.js]:
```suggestion
metatests.test('with an empty array', test => {
```

File: test/array.findIndex.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    (el, callback) => process.nextTick(() => callback(null, true)),
+    (err, index) => {
+      test.error(err);
+      test.strictSame(index, -1);
+      test.end();
+    }
+  );
+});
+
+metatests.test('with array without element which is searching', test => {
```
Comment (DzyubSpirit) [inline test/array.findIndex.js]:
```suggestion
metatests.test('with an array not containing the searched element', test => {
```

File: lib/array.js [hunk 5]
```diff
@@ -416,25 +443,17 @@ const find = (items, fn, done) => {
...
-      return;
+  findIndex(
+    items,
+    (item, callback) => fn(item, (err, accepted) => callback(err, !accepted)),
+    (err, index) => {
+      if (err || index !== -1) {
+        done(err, err ? undefined : false);
+        return;
+      }
+      done(null, true);
```
Comment (DzyubSpirit) [inline lib/array.js]:
```suggestion
      done(null, index !== -1);
```
And then it's possible to simplify the check upper to 
```
if (err) {
  done(err);
  return;
}
```

File: lib/array.js [hunk 6]
```diff
@@ -448,30 +467,13 @@ const every = (items, fn, done) => {
...
-        return;
-      }
-      i++;
-      next();
-    });
-  };
-
-  if (len > 0) next();
-  else done(null, false);
+    done(null, true);
```
Comment (DzyubSpirit) [inline lib/array.js]: ditto

**Comment by mapogolions** [comment]
Thanks, I really appreciate it. I found a problem that could affect the future of this pull request (#442). I will resolve all conversions above as soon as this issue is resolved

**Pull request closed without merging**